### PR TITLE
[RAPTOR-8303] tests refactoring

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
1. Up to today integration tests were run in a dedicated test container.
- Now: ML specific cases (sklearn, keras, etc, R, java) are run in the related pudlic env. Tests are managed by `test_integration_per_framework` scripts
2. The rest of cases, general cases, use sklearn and are run natively in jenkins by `test_integration_general`. 
3. Remove test image docker context
4. Remove an image used to test `--docker` option. Use `python3_sklearn` env instead

### CODE REVIEW

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container test_integration_r_environment

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container test_integration_r_environment

**Comment by yakov-g** [comment]
jenkins run test_integration_r_environment

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_without_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_container

**Comment by yakov-g** [comment]
jenkins run test_integration_without_container

**Comment by yakov-g** [comment]
jenkins run test_integration_without_container

**Comment by yakov-g** [comment]
jenkins run test_integration_without_container

**Comment by yakov-g** [comment]
jenkins run test_integration_without_container

**Comment by yakov-g** [comment]
jenkins run test_integration_without_container

**Comment by yakov-g** [comment]
jenkins run test_integration_without_container

**Comment by yakov-g** [comment]
jenkins run test_integration_without_container

**Comment by yakov-g** [comment]
jenkins run this

**Review by zohar-mizrahi — COMMENTED** [review]
In the future, please make sure to split such big PR into smaller ones :)

File: tests/conftest.py [hunk 1]
```diff
@@ -879,68 +879,68 @@ def __init__(self):
 
 @pytest.fixture(scope="session")
 def get_dataset_filename():
```
[the remark this answers is not in the document]
Comment (yakov-g) [reply tests/conftest.py]: DONE

File: jenkins/test_integration_general.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+title "DEBUG: print java version"
+java -version
+javac -version
+
+pip install -U pip
+pip install pytest pytest-runner pytest-xdist retry scikit-learn==0.24.2
+
+# > NOTE: when pinning datarobot-mlops to 8.2.1 and higher you may need to reinstall datarobot package
+# as datarobot-mlops overwrites site-packages/datarobot. [AGENT-3504]
+pip install datarobot-mlops==8.2.7
```
[the remark this answers is not in the document]
Comment (yakov-g) [reply jenkins/test_integration_general.sh]: I put it into `requirements_test` for now

File: jenkins/test_integration_general.sh [hunk 3]
```diff
@@ -25,11 +25,15 @@ java -version
 javac -version
 
 pip install -U pip
-pip install pytest pytest-runner pytest-xdist retry scikit-learn==0.24.2
+#pip install pytest pytest-runner pytest-xdist retry scikit-learn==0.24.2
```
Comment (zohar-mizrahi) [inline jenkins/test_integration_general.sh]: I'd encourage removing this kind of comments at all.

File: jenkins/test_integration_general.sh [hunk 4]
```diff
@@ -25,11 +25,15 @@ java -version
...
+# test custom java predictor fails if these packages are installed before
+title "Installing requirements for all the tests:  ${GIT_ROOT}/requirements_test.txt"
+pip install -r ${GIT_ROOT}/requirements_test.txt
+
 
 # > NOTE: when pinning datarobot-mlops to 8.2.1 and higher you may need to reinstall datarobot package
 # as datarobot-mlops overwrites site-packages/datarobot. [AGENT-3504]
-pip install datarobot-mlops==8.2.7
+# pip install datarobot-mlops==8.2.7
```
Comment (zohar-mizrahi) [inline jenkins/test_integration_general.sh:34]: I'd encourage removing such comments.

**Review by zohar-mizrahi — APPROVED** [review]

**Review by akshoop — COMMENTED** [review]
Skimmed through changes and they seem to make sense, PR checks of the new jobs are green, this is a large PR though so need help from other Core Modeling teammate to review these tests setup changes, CC @eric-s-s

**Review by scottp-ml — APPROVED** [review]

**Comment by yakov-g** [comment]
> Skimmed through changes and they seem to make sense, PR checks of the new jobs are green, this is a large PR though so need help from other Core Modeling teammate to review these tests setup changes, CC @eric-s-s

thank you

**Pull request merged**